server/routes/api/collections/collections.ts

```diff
@@ -881,6 +892,188 @@ router.post(
+    const { transaction } = ctx.state;
+    const { id } = ctx.input.body;
+    const { user } = ctx.state.auth;
+
+    const collection = await Collection.scope([
+      {
+        method: ["withMembership", user.id],
+      },
+      "withArchivedBy",
+    ]).findByPk(id, {
+      transaction,
+    });
+
+    if (!collection) {
+      throw NotFoundError("Collection not found");
+    }
+
+    authorize(user, "archive", collection);
+
+    collection.archivedAt = new Date();
+    collection.archivedById = user.id;
+    await collection.save({ transaction });
+    await collection.reload({ transaction });
+
+    // Archive all documents within the collection
+    await Document.update(
+      {
+        lastModifiedById: user.id,
+        archivedAt: collection.archivedAt,
+      },
+      {
+        where: {
+          teamId: collection.teamId,
+          collectionId: collection.id,
+          archivedAt: {
+            [Op.is]: null,
+          },
+        },
+        transaction,
+      }
+    );
+
+    await Event.createFromContext(
+      ctx,
+      {
+        name: "collections.archive",
+        collectionId: collection.id,
+        data: {
+          name: collection.name,
+          archivedAt: collection.archivedAt,
+        },
+      },
+      { transaction }
+    );
+
+    ctx.body = {
+      data: await presentCollection(ctx, collection),
+      policies: presentPolicies(user, [collection]),
+    };
+  }
+);
+
+router.post(
+  "collections.restore",
+  auth(),
+  validate(T.CollectionsRestoreSchema),
+  transaction(),
+  async (ctx: APIContext<T.CollectionsRestoreReq>) => {
+    const { transaction } = ctx.state;
+    const { id } = ctx.input.body;
+    const { user } = ctx.state.auth;
+
+    const collection = await Collection.scope({
+      method: ["withMembership", user.id],
+    }).findByPk(id, {
+      transaction,
+    });
+
+    if (!collection) {
+      throw NotFoundError("Collection not found");
+    }
+
+    authorize(user, "restore", collection);
+
+    const collectionArchivedAt = collection.archivedAt;
+
+    await Document.update(
+      {
+        lastModifiedById: user.id,
+        archivedAt: null,
+      },
+      {
+        where: {
+          collectionId: collection.id,
+          teamId: user.teamId,
+          archivedAt: collection.archivedAt,
+        },
+        transaction,
+      }
+    );
+
+    collection.archivedAt = null;
+    collection.archivedById = null;
+    await collection.save({ transaction });
+
+    await Event.createFromContext(
+      ctx,
+      {
+        name: "collections.restore",
+        collectionId: collection.id,
+        data: {
+          name: collection.name,
+          archivedAt: collectionArchivedAt,
+        },
+      },
+      { transaction }
+    );
+
+    ctx.body = {
+      data: await presentCollection(ctx, collection!),
+      policies: presentPolicies(user, [collection]),
+    };
+  }
+);
+
+router.post(
+  "collections.archived",
```

Comment:
Maybe we can get rid of this in favor or `collections.list` with `StatusFilter`, similar to `documents.list`?

cc: @tommoor

Reply:
Don't mind taking this as a follow up though. Helps cut down further work here – lots of small tasks really add up 🙈

Reply:
As it's brand new though… it should reduce code overall